app/code/Magento/Ui/view/base/web/js/grid/columns/image-preview.js

```diff
@@ -174,6 +187,36 @@ define([
                 block: 'center',
                 inline: 'nearest'
             });
-        }
+        },
+
+        /**
+         * Set image preview keyboard navigation listener
+         */
+        setNavigationListener: function () {
+            var imageIndex, endIndex, key,
```

Comment:
I don't think start and endIndex should be handled here. They should be handled in next and prev methods (if it is currently possible to trigger them for first/last image)

Reply:
Fixed this too, thank you for your guidance!